exaudfclient/base/python/python3/python_ext_dataframe.cc

```diff
@@ -856,19 +862,20 @@ inline void handleEmitPyBool(
         PyPtr& pyValue,
         PyPtr& pyResult,
         PyPtr& pySetNullMethodName){
-    PyPtr pyBool(PyList_GetItem(columnArrays[c].get(), r));
-    checkPyPtrIsNull(pyBool);
-    if (isNoneOrNA(pyBool.get())) {
+    //PyList_GetItem returns a 'borrowed' reference. Must not call XDECREF() on it.
+    PyObject *pyBool = PyList_GetItem(columnArrays[c].get(), r);
+    checkPyObjIsNotNull(pyBool);
+    if (isNoneOrNA(pyBool)) {
         pyResult.reset(PyObject_CallMethodObjArgs(resultHandler, pySetNullMethodName.get(), pyColSetMethods[c].first.get(), NULL));
         return;
     }
     switch (colInfo[c].type) {
         case SWIGVMContainers::BOOLEAN:
-            if (pyBool.get() == Py_True) {
+            if (pyBool == Py_True) {
```

Comment:
My understanding is that `Py_True` (which comes from the Python header file) is a static pointer to the (immortal) "True" object of Python. 
See https://docs.python.org/3/c-api/bool.html#c.Py_True

Reply:
The thing is, what ever pyBool is, it is either Py_True or Py_False, otherwise we would need to use a Python comparison instead of a c++. That means, the comparison is meaningless, because we do both cases the same. We could replace this with INCREF(pyBool)